packages/avocode-email-tagsinput/test/e2e/basic.spec.js

```diff
@@ -0,0 +1,116 @@
+  assertEmailTagCount,
+  assertInputText,
+} from './test-utils'
+
+describe('Basic Email Tags Input', () => {
+  let browser
+  let page
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch(browserOptions)
+  })
+
+  beforeEach(async () => {
+    page = await browser.newPage()
+    await page.goto(URL)
+  })
+
+  afterEach(async () => {
+    await page.close()
+  })
+
+  afterAll(async () => {
+    await browser.close()
+  })
+
+  it('should render editor', async () => {
+    const editor = await page.waitForSelector(tagsInputFieldSelector)
+    expect(editor).to.exist
+  })
+
+  describe('user typing', () => {
+    it('should insert text', async () => {
+      await page.click(tagsInputFieldSelector)
+      await page.type(tagsInputFieldSelector, 'abcd')
+      await assertInputText(page, 'abcd')
+    })
+
+    it('should delete text with backspace', async () => {
+      await page.click(tagsInputFieldSelector)
+      await page.type(tagsInputFieldSelector, 'abcd')
+      await page.keyboard.press('Backspace')
+      await assertInputText(page, 'abc')
+    })
+
+    it('should delete text with delete', async () => {
+      await page.click(tagsInputFieldSelector)
+      await page.type(tagsInputFieldSelector, 'abcd')
+      await page.keyboard.press('ArrowLeft')
+      await page.keyboard.press('Delete')
+      await assertInputText(page, 'abc')
+    })
+
+    it('should navigate one char left', async () => {
+      await page.click(tagsInputFieldSelector)
+      await page.type(tagsInputFieldSelector, 'abcd')
+      await page.keyboard.press('ArrowLeft')
+      const [ anchorOffset ] = await getAnchorAndFocus(page)
+      expect(anchorOffset).to.equal(3)
+    })
+  })
+
+  describe('user typing different email edge cases', () => {
+    const unsupportedEmailEdgeCases = [
+      'testemail',
+      'testemail@ex',
+      'testemail@ex.11',
+      'double"123@quotes.co',
+      'withb[]g@bracket.com',
+      'withsm()ll@bracket.com',
+    ]
+    const supportedEmailEdgeCases = [
+      '1234@test.co',
+      'testemail@11.ex',
+      "single'123@quote.co",
+      'withcu{}ly@bracket.co',
+      'सिमप्लेफोल्देर$%#?&@nepali.co',
+    ]
+
+    for (const testEmail of unsupportedEmailEdgeCases) {
+      it(`should not be allowed to add ${testEmail} as an email tag`, async () => {
+        await addEmailTag(page, testEmail)
+        await assertEmailTagCount(page, 0)
+      })
+    }
+
+    for (const testEmail of supportedEmailEdgeCases) {
+      it(`should be allowed to add ${testEmail} as an email tag`, async () => {
+        await addEmailTag(page, testEmail)
+        await assertEmailTagCount(page, 1)
+      })
+    }
+
+    it('spaces provided inside email are omitted while creating email tag', async () => {
+      await addEmailTag(page, 'em a il@s pa ce. co')
+      await assertInputText(page, 'email@space.co')
+    })
+  })
+
+  describe('user can add new tag typing valid email and hitting comma, enter or space', () => {
```

Comment:
OK this is similar point to dynamic specs above so the same applies here.

It's even more evident here. It'd be better to be verbose here, I know you're trying to do things "smart" here but my opinion is it's better to be verbose in specs (from my experience).

I think it's better to have a specific spec saying `it should be possible to ad dan email tag hitting comma key` and then test for that specific key.